mongo-executer.js

```diff
@@ -1,113 +1,285 @@
+ * Instead it just returns the result `{n: Number}`.
+ * @param {Object} cursor
+ */
+const getResult = (cursor) => {
+    // Mongo 3.0.x support
+    if (cursor.result) {
+        return F.of(cursor.result);
+    }
+    // Mongo 3.1.x support
+    if (cursor.toArray) {
+        return F.node(done => cursor.toArray(done));
+    }
 
-const executeFind = R.curry(function(collection, query, projection, options, db) {
+    // mongo-mock support
+    return F.of(cursor);
+};
 
-    const findInCollection = () => function(callback)  {
-        return db.collection(collection).find(query, projection, callback)
-    }
+const executeFind = R.curry((collection, query, projection, options, db) => {
+    const findInCollection = () => db.collection(collection).find(query).project(projection);
+
+    const rejector = rejectMongoOf(collection, query, projection);
+
+    return F.try(findInCollection)
+        .chainRej(rejector)
+        .chain(toArray(options.skip, options.limit));
+});
+
+const buildFind = (collection, query, projection, skip, limit) => executeFind(
+    collection,
+    query,
+    projection,
+    { skip, limit },
+);
+
+const executeUpdate = R.curry((collection, query, object, upsert, db) => {
+    const updatedObject = updateObject(object);
+    const updateCollection = () => db.collection(collection).updateMany(
+        query,
+        updatedObject,
+        { upsert },
+    );
+
+    const rejector = rejectMongoOf(collection, query, null, updatedObject);
+
+    return F.tryP(updateCollection)
+        .chainRej(rejector)
+        .chain(getResult);
+});
 
-    const rejector = rejectMongoOf(collection, query, projection)
+const executeUpdateOne = R.curry((collection, query, object, upsert, db) => {
+    const updateCollection = () => db.collection(collection).updateOne(query, object, { upsert });
 
-    return F.node(
-        findInCollection(query, projection)
-    ).chainRej(
-        rejector
-    ).chain(
-        toArray(options.skip, options.limit)
-    ) ;
-})
+    const rejector = rejectMongoOf(collection, query, null, object);
 
+    return F.tryP(updateCollection)
+        .chainRej(rejector)
+        .chain(getResult);
+});
 
-const buildFind = function(collection, query, projection, skip, limit){
-    return executeFind(collection, query, projection, {skip, limit})
-}
+const executeUpdateMany = R.curry((collection, query, object, upsert, db) => {
+    const updateCollection = () => db.collection(collection).updateMany(query, object, { upsert });
 
-const executeUpdate = R.curry(function(collection, query, object, upsert, db) {
+    const rejector = rejectMongoOf(collection, query, null, object);
 
-    const updateCollection = () => function(callback)  {
-        return db.collection(collection).update(query, object, {"upsert": upsert}, callback)
+    return F.tryP(updateCollection)
+        .chainRej(rejector)
+        .chain(getResult);
+});
+
+const executeInsert = R.curry((collection, object, db) => {
```

Comment:
Mark deprecated functions with a comment so folks are aware of what to use go-forward.